worlds/osrs/__init__.py

```diff
@@ -0,0 +1,616 @@
+        else:
+            cooking_tasks.reverse()
+        cooking_tasks = cooking_tasks[0:self.multiworld.max_cooking_tasks[self.player]]
+        cooking_weight = self.multiworld.cooking_task_weight[self.player] if len(cooking_tasks) > 0 else 0
+
+        firemaking_tasks = [task for task in self.locations_by_category["Firemaking"]
+                            if task.skills[0].level <= int(self.multiworld.max_firemaking_level[self.player])]
+        if not self.multiworld.progressive_tasks[self.player]:
+            rnd.shuffle(firemaking_tasks)
+        else:
+            firemaking_tasks.reverse()
+        firemaking_tasks = firemaking_tasks[0:self.multiworld.max_firemaking_tasks[self.player]]
+        firemaking_weight = self.multiworld.firemaking_task_weight[self.player] if len(firemaking_tasks) > 0 else 0
+
+        woodcutting_tasks = [task for task in self.locations_by_category["Woodcutting"]
+                             if task.skills[0].level <= int(self.multiworld.max_woodcutting_level[self.player])]
+        if not self.multiworld.progressive_tasks[self.player]:
+            rnd.shuffle(woodcutting_tasks)
+        else:
+            woodcutting_tasks.reverse()
+        woodcutting_tasks = woodcutting_tasks[0:self.multiworld.max_woodcutting_tasks[self.player]]
+        woodcutting_weight = self.multiworld.woodcutting_task_weight[self.player] if len(woodcutting_tasks) > 0 else 0
+
+        all_tasks = [
+            combat_tasks,
+            prayer_tasks,
+            magic_tasks,
+            runecraft_tasks,
+            crafting_tasks,
+            mining_tasks,
+            smithing_tasks,
+            fishing_tasks,
+            cooking_tasks,
+            firemaking_tasks,
+            woodcutting_tasks,
+            general_tasks
+        ]
+        all_weights = [
+            combat_weight,
+            prayer_weight,
+            magic_weight,
+            runecraft_weight,
+            crafting_weight,
+            mining_weight,
+            smithing_weight,
+            fishing_weight,
+            cooking_weight,
+            firemaking_weight,
+            woodcutting_weight,
+            general_weight
+        ]
+
+        while locations_added < locations_required:
+            taskChosen = False
+            if all_tasks:
+                chosen_task = rnd.choices(all_tasks, all_weights)[0]
+                if len(chosen_task) > 0:
+                    task = chosen_task.pop()
+                    taskChosen = True
+                if len(chosen_task) == 0:
+                    index = all_tasks.index(chosen_task)
+                    del all_tasks[index]
+                    del all_weights[index]
+
+            else:
+                assert len(general_tasks) > 0, "There are not enough avaialbe tasks to vfill the remaining pool for OSRS"
+                task = general_tasks.pop()
+                pass
+            if taskChosen:
+                self.add_location(task)
+                locations_added += 1
```

Comment:
Not gonna repeat myself over and over, but once again, you could reduce nesting using guard conditions with `continue` and do `if some_list` instead of `if len(some_list) > 0`

In fact, if you use `continue` to exit out of code paths where no task was chosen, you can get rid of the ugly `taskChosen` variable.